### PR TITLE
time-util: clock mapping improvements

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by poettering** [comment]
This is pretty much a follow-up for #16428, hence /cc @filbranden

File: src/core/timer.c [hunk 1]
```diff
@@ -388,12 +387,14 @@ static void timer_enter_waiting(Timer *t, bool time_change) {
...
-                        if (v->next_elapse < dts.realtime)
-                                v->next_elapse = dts.realtime;
+                        /* To make the delay due to RandomizedDelaySec= work even at boot, if the scheduled
+                         * time has already passed, set the time when systemd first started as the scheduled
+                         * time. Note that we base this on the monotonic timestamp of the boot, not the
+                         * realtime one, since the wallclock might have been off during boot. */
+                        rebased = map_clock_usec(UNIT(t)->manager->timestamps[MANAGER_TIMESTAMP_USERSPACE].monotonic,
+                                                 CLOCK_MONOTONIC, CLOCK_REALTIME);
+                        if (v->next_elapse < rebased)
+                                v->next_elapse = rebased;
```
Comment (keszybz) [inline src/core/timer.c:397]: Yep, this is much clearer now.

**Comment by keszybz** [comment]
bionic-i386 failed in boot-and-services, but I have no idea why exactly.

**Comment by keszybz** [comment]
I restarted github ci.

**Pull request merged**